lib/src/rules/always_require_non_null_named_parameters.dart

```diff
@@ -88,16 +88,34 @@ class Visitor extends SimpleAstVisitor {
     if (parent is FunctionExpression) {
       _checkParams(params, parent.body);
     } else if (parent is ConstructorDeclaration) {
+      _checkInitializerList(params, parent.initializers);
       _checkParams(params, parent.body);
     } else if (parent is MethodDeclaration) {
       _checkParams(params, parent.body);
     }
   }
 
-  _checkParams(List<DefaultFormalParameter> params, FunctionBody body) {
+  void _checkInitializerList(List<DefaultFormalParameter> params,
+      NodeList<ConstructorInitializer> initializers) {
+    final asserts = initializers
+        .where((i) => i is AssertInitializer)
+        .map((e) => e as AssertInitializer)
+        .map((e) => e.condition)
```

Comment:
It probably doesn't matter for lists this short, so no action required, but there is a performance difference between two map operations and one:
```
.map((e) => (e as AssertInitializer).condition)
```
For a list with 2000 items, and using one `where`, one or two `map`s and one `toList`, I'm measuring 7 ms for two and 2 ms for one.